### ISSUE TITLE
Chords returned by Scale.scaleChords

### ISSUE THREAD

**Issue by colshacol** [tonaljs/tonal#206]
![image](https://user-images.githubusercontent.com/19484365/91179333-fb694e00-e6ab-11ea-8c9e-c6f961c33404.png)

I am confused as to how to work with this. I was expecting the output to be more along the lines of the actual chords in the scale, i.e: Cmaj, Dmin, Emin, Fmaj, Gmaj, Amin, Bdim, Cmaj7, etc...

Any advice on how I can get the data I am looking for?

**Comment by KaipoJames** [comment]
You need to use the 'key' module to get the data you are looking for. 
For ES6 syntax:

**import { Key } from "@tonaljs/tonal";**

**const keyChords = Key.majorKey("C");
###console.log(keyChords.chords);**

**Output:
["Cmaj7", "Dm7", "Em7", "Fmaj7", "G7", "Am7", "Bm7b5"]``**


Documentation on the tonal-key package:
https://www.npmjs.com/package/tonal-key

Hope this helps! :)

**Comment by colshacol** [comment]
Ah, of course. My apologies! I am a musician of sorts, but for some reason my developer mind took the Key module to be related to the Note module. (As in piano keys. How foolish of me. -_-)

A few things, though:

1. The `Key.chords` method provides seventh chords. Maybe the naming could be more intuitive?

`Key.seventhChords("A major")`?

(I'd be happy to provide a PR if it is discussed and deemed acceptable.)

I see the `triads` method, too, and see that these two methods are the solution for my issue here, so thank you for the reference!

2. A little off topic of code and kind of into music theory, but am correct to assume that if I take the `Key.chords`, `Key.triads`, and also duplicate them each with variations using the output from my original post, I would have all possible chords/variations for my scale?

**Comment by colshacol** [comment]
Ah, alright! I missed the `Key` package. 

`Key.majorKey(...).chords` gives you just the 7th chords, though.

I noticed that the `tonal-key` package gives you the `.triads` method, so combining these two (or resorting to simply using `tonal-key`  methods `chords` and `triads`) I can get a good set of chords from a scale, such as: `["C", "Dm", "Em", "F", "G", "Am", "Bdim", "CMaj7", "Dm7", "Em7", "FMaj7", "G7", "Am7", "Bm7b5"]`

Any advice on how I should be applying the original output to these chords to get the full list of chords/variants that would fit in my key/scale? `["5", "64", "M", "M13", "M6", "M69", "M7add13", "M7sus4", "M9", "M9sus4", "Madd9", "Maj7", "Msus2", "Msus4", "sus24"]`

**Comment by KaipoJames** [comment]
Yes good find about the triads in the tonal-key package! I also am a musician so I was confused too between the scale and key packages! 😂 I love music theory so this package is very interesting to me!

Also could you provide a specific example for what you are wanting to access in your last paragraph? It is the part where you mention about getting the full list of chords/variants in your key/scale.
Sorry, I just want to make sure I understand fully what you are asking so I can try to help! 😎

**Comment by colshacol** [comment]
I am currently running a puppeteer script to scrape the scale/chord associations from some outdated website. 😋 What I needed to get from tonal is, for example, is the following chords when I input "A major":

`[
      "C#m",
      "D",
      "E",
      "F#m",
      "G#dim",
      "A",
      "Bm",
      "C#sus4",
      "Dsus2",
      "Esus4",
      "Esus2",
      "F#sus4",
      "F#sus2",
      "Asus4",
      "Asus2",
      "Bsus4",
      "Bsus2",
      "C#m7",
      "C#7sus4",
      "D6",
      "Dmaj7",
      "E6",
      "E7",
      "E7sus4",
      "E7sus2",
      "F#m7",
      "F#7sus4",
      "F#7sus2",
      "G#m7b5",
      "A6",
      "Amaj7",
      "Bm6",
      "Bm7",
      "B7sus4",
      "B7sus2",
      "C#m11",
      "D6/9",
      "Dmaj9",
      "Dmaj13",
      "E6/9",
      "E9",
      "E13",
      "E9sus4",
      "E9sus2",
      "F#m9",
      "F#m11",
      "F#9sus4",
      "F#9sus2",
      "A6/9",
      "Amaj9",
      "Amaj13",
      "Bm9",
      "Bm11",
      "Bm13",
      "B9sus4",
      "B9sus2"
    ]`

Basically, I need to be able to access _every_ chord and variation that fits in a key/scale.

**Comment by colshacol** [comment]
[Here](https://github.com/colshacol/key-scale-chords/blob/master/scrapedData.json) is the data I scraped. The source seemed pretty trustworthy, albeit outdated. 🤷‍♂️

**Comment by KaipoJames** [comment]
Ok I see what you need. I am not too sure at the moment how to do that but I will research it more and hopefully we can find a good solution!

**Comment by colshacol** [comment]
Since you seem to know a little more about music theory than I do, let me you ask you this:

If, for example, C major contains the notes C, D, E, F, G, A, and B, would any chord I composed out of these notes fit in the scale?

**Comment by KaipoJames** [comment]
Yes, that is correct. As long as all of the notes in a given chord fit into a scale, then the chord fits in that scale.

**Comment by danigb** [comment]
Hello!

I think you are mixing several things here. 

Each key naturally have some chords. You can find them using the Key module  👉  https://github.com/tonaljs/tonal/tree/master/packages/key 

The `majorKey` function returns an object with all chords (but under different keys). You have to merge them all. For example:

```js
import { Key } from "@tonaljs/tonal";

const majorC = Key.majorKey('C')
const allChords =  [
  ...majorC.chords, 
  ...majorC.secondaryDominants, 
  ...majorC.secondaryDominantsMinorRelative, 
  ...majorC.substituteDominants, 
  ...majorC.substituteDominantsMinorRelative
].filter(x => x)
console.log(allChords)
```

Outputs:
```
[
  'Cmaj7', 'Dm7',  'Em7',   'Fmaj7',
  'G7',    'Am7',  'Bm7b5', 'A7',
  'B7',    'C7',   'D7',    'E7',
  'Em7b5', 'F#m7', 'Gm7',   'Am7',
  'Bm7b5', 'Eb7',  'F7',    'Gb7',
  'Ab7',   'Bb7',  'Em7',   'Cm7',
  'Dbm7',  'Am7',  'Fm7'
]
```

But also each Key has several scales (or modes).

Lot of chords can "fit" an scale and that's when `scaleChords` function come in:

```js
Scale.scaleChords('C major') // => 
[
  '5',      'sus4',
  'M7sus4', 'M',
  'maj7',   '6',
  'sus2',   'sus24',
  'M9sus4', 'Madd9',
  'maj9',   '6/9',
  'maj13',  'M7add13'
]
```

That result means that "C5", "Csus4", "CM7sus4", ... chords fits on C major scale. 

You could apply the same operation to every scale on the key.

**Comment by ppciesiolkiewicz** [comment]
@danigb 
Is it possible to get chords for any scale returned by ` Scale.names()`, let's say `egyptian` or `iwato`. How could I know whether the scale is minor or major? Or am I misunderstanding something here?

**Comment by jabza** [comment]
Regarding this, when trying `scaleChords` on C minor, I noticed 'M' is not returned, but C minor does include major chords, possible bug, or am I misunderstanding this?

```
Scale.scaleChords('C minor') // => 
[
  '5',     '7#5sus4', 'sus4',
  '7sus4', 'm#5',     'm7#5',
  'm',     'm7',      '4',
  'madd4', 'm7add11', 'sus2',
  'sus24', '11',      '9sus4',
  'm9#5',  'madd9',   'm9',
  'm11A',  'm11'
]
```

Cheers
